openshift_day2_configurator/configurators/nodes.py

```diff
@@ -0,0 +1,613 @@
+
+
+def configure_journald_setting_on_nodes_message(nodes_type: str) -> str:
+    return f"Configure journald settings on {nodes_type} nodes"
+
+
+def configure_image_registry_setting_on_nodes_message(nodes_type: str) -> str:
+    return f"Configure image registry settings on {nodes_type} nodes"
+
+
+def adding_extensions_to_rhcos_on_nodes_message(nodes_type: str) -> str:
+    return f"Adding extensions to RHCOS on {nodes_type} nodes"
+
+
+def loading_custom_firmware_blobs_on_nodes_message(nodes_type: str) -> str:
+    return f"Loading custom firmware blobs in the machine config manifest for {nodes_type} nodes"
+
+
+def get_machine_config_pool(client: DynamicClient, nodes_type: str) -> MachineConfigPool:
+    return [mcp for mcp in MachineConfigPool.get(dyn_client=client) if mcp.name == nodes_type][0]
+
+
+def get_number_of_nodes(client: DynamicClient, nodes_type: str) -> int:
+    return len([
+        node for node in Node.get(dyn_client=client) if f"node-role.kubernetes.io/{nodes_type}" in node.labels.keys()
+    ])
+
+
+def generate_firmware_machine_config_file(
+    firmware_package_name: str,
+    firmware_files_dir: str,
+    butane_content: str,
+) -> str:
+    firmware_machine_config_file: str = f"{firmware_package_name}.yaml"
+
+    run_command(
+        command=shlex.split(f"butane -o {firmware_machine_config_file} --files-dir {firmware_files_dir}"),
+        input=butane_content,
+    )
+
+    return firmware_machine_config_file
+
+
+def wait_for_machine_config_pool_to_update(
+    client: DynamicClient,
+    nodes_type: str,
+    node_operation_message: str,
+    error_message: str,
+    machine_config_name: str,
+) -> Dict[str, Dict[str, Union[str, bool]]]:
+    nodes_timeout = NODE_TIMEOUT_7MIN * get_number_of_nodes(client=client, nodes_type=nodes_type)
+
+    try:
+        for mcp_sample in TimeoutSampler(
+            wait_timeout=nodes_timeout,
+            sleep=10,
+            func=get_machine_config_pool,
+            client=client,
+            nodes_type=nodes_type,
+        ):
+            if machine_config_name in [
+                config_source["name"]
+                for config_source in mcp_sample.instance.to_dict()["spec"]["configuration"]["source"]
+            ]:
+                mcp_sample.wait_for_condition(
+                    condition=MachineConfigPool.Status.UPDATED,
+                    status=MachineConfigPool.Condition.Status.TRUE,
+                    timeout=nodes_timeout,
+                )
+                return {
+                    node_operation_message: {
+                        "res": True,
+                        "err": "",
+                    }
+                }
+
+    except Exception as ex:
+        return {
+            node_operation_message: {
+                "res": False,
+                "err": f"{error_message}: {ex}",
+            }
+        }
+
+    return {
+        node_operation_message: {
+            "res": False,
+            "err": error_message,
+        }
+    }
+
```

Comment:
the function either returns the dict under the try or under the exception; when does this code is reached?

Reply:
never, though it raises a mypy error:
```openshift_day2_configurator/configurators/nodes.py:80: error: Missing return statement```

should we add `# type: ignore` on this?